lib/js/jslib.js

```diff
@@ -1,187 +1,255 @@
 'use strict';
 
 /* LIST MANIPULATING FUNCTIONS */
+const LINKEDLIST =  new function(){
```

Comment:
Please avoid using `new function` as it suffers from potential security and performance issues. It is somewhat similar to be using `eval` to dynamically create and execute arbitrary code. Rather use 
```
const LINKEDLIST = (function() {
   return ....;
}());
```